### PR TITLE
Feature/s3 advanced dynamic search

### PR DESCRIPTION
**Pull request by danielanthonyr**
pull request for advanced dynamic search, now includes documentation

### CODE REVIEW

File: app/src/main/java/Fragments/HomeFragment.java [hunk 1]
```diff
@@ -139,19 +136,21 @@ public boolean onQueryTextSubmit(String query) {
 				}
 
 			    @Override
+				// Responsible for displaying all possible string from the list based on each additionnal character input made by user
 			    public boolean onQueryTextChange(String newText) {
```
Comment (LHuss) [inline app/src/main/java/Fragments/HomeFragment.java:147]: Are you aware that you changed the functionality of the search we had from only applying the search filter when the user is done to actively scanning through every item EVERY time a change is made? 

The changes you made are inherently voiding the "onQueryTextSubmit" functionality above.

**Review by LHuss — CHANGES_REQUESTED** [review]
Aside from my question, everything looks fine

**Comment by LHuss** [comment]
Also, please be sure to update your branch from master.

**Comment by danielanthonyr** [comment]
@LHuss branch now updated with master, also we will keep implementation the same for now, we can remove down the line if too many listem items and if it starts to hinder performance we will remove.

**Review by Alex-Hayeur — COMMENTED** [review]
Aside from the concerns brought up by Loic, the changes seem good. I appreciate what was done to make the code cleaner.

File: app/src/main/java/Fragments/HomeFragment.java (hunk 1 above)
Comment (kkhh001) [reply app/src/main/java/Fragments/HomeFragment.java:147]: Taking note of that 👍 . Will make necessary changes next sprint

**Review by kkhh001 — APPROVED** [review]
Other than Loïc's comment/review, I don't see any other objection!

**Review by LHuss — APPROVED** [review]

**Pull request merged**